lib/plugins/multiEntity.js

```diff
@@ -184,13 +184,40 @@ function generateNewCEsNgsi2(entity, newEntities, entityTypes, typeInformation,
         return result;
     }
 
+    function filterByAttributeObjectIds() {
+        var result = {};
+        for (var att in entity) {
+            if (entity.hasOwnProperty(att)) {
+                if (_.contains(newEntityAttributeNames, att)) {
+                    if (entity[att].object_id && _.contains(newEntityAttributeObjectIds, entity[att].object_id )){
+                        result[att] = entity[att];
+                        delete entity[att].object_id;
+                    } else {
+                        // Check matches in rest of multientity attributes with same name (#635)
+                        if (entity[att].multi) {
+                            for (var j in entity[att].multi) {
+                                if (entity[att].multi[j].object_id && _.contains(newEntityAttributeObjectIds, entity[att].multi[j].object_id)) {
+                                    result[att] = entity[att].multi[j];
+                                    delete entity[att].multi[j].object_id;
+                                }
+                            }
+                            delete entity[att].multi;
+                        }
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
     var attsArray = utils.extractAttributesArrayFromNgsi2Entity(entity);
     ctx = parser.extractContext(attsArray);
 
     for (var i = 0; i < newEntities.length; i++) {
         newEntityAttributeNames = _.pluck(multiEntityAttributes.filter(filterByEntityName(newEntities[i])), 'name');
         newEntityAttributeObjectIds = _.pluck(multiEntityAttributes.filter(filterByEntityName(newEntities[i])), 'object_id');
-        newEntityAttributes = filterAttributes();
+        //newEntityAttributes = filterAttributes();
```

Comment:
Leftover comment? Should be removed?

Reply:
fixed in a79ffd8438cadf1fd25a5cc22e3e49ec8170a437